python/mxnet/gluon/text/vocab.py

```diff
@@ -321,3 +324,17 @@ def to_tokens(self, indices):
                 tokens.append(self._idx_to_token[idx])
 
         return tokens[0] if to_reduce else tokens
+
+    def __repr__(self):
+        return 'Vocab(size={}, unk="{}", reserved="{}")'.format(len(self), self._unknown_token,
+                                                                self._reserved_tokens)
+
+    def json_serialize(self):
+        if self._embedding is not None:
+            warnings.warn('Serialization of attached embedding is not supported. Please save separately')
+        vocab_dict = {}
+        vocab_dict['idx_to_token'] = self._idx_to_token
+        vocab_dict['token_to_idx'] = self._token_to_idx
+        vocab_dict['reserved_tokens'] = self._reserved_tokens
+        vocab_dict['unknown_token'] = self._unknown_token
+        return json.dumps(vocab_dict)
```

Comment:
Vocab serialization is needed to accompany pretrained models. Otherwise weights won't match.